### PR TITLE
undo a delete

### PR DESCRIPTION
**Pull request by cdubthecoolcat**

### CODE REVIEW

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java [hunk 1]
```diff
@@ -77,7 +77,6 @@ public boolean deleteConnection(int id) {
         SQLiteDatabase database = getWritableDatabase();
         String idFilter = ID_COL + "=" + "'" + id + "'";
         database.delete(CONNECTIONS_TABLE, idFilter, null);
-        database.close();
```
Comment (tylerbwong) [inline app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java:80]: Any reason why this was removed?

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java [hunk 2]
```diff
@@ -100,6 +99,28 @@ public void updateDate(int id, String date) {
         });
     }
 
+    public List<String> getConnection(int id) {
```
Comment (tylerbwong) [inline app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java]: Should this be returning a `Connection` object instead of a `List` of its fields?

Comment (tylerbwong) [reply app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java]: This function should probably be async, so you can wrap it in a `Single`. So the method would look like:
```java
public Single<Connection> getConnection(int id) { 
    return Single.fromCallable(() -> {
        ...
    })
}
```

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java (hunk 1 above)
Comment (cdubthecoolcat) [reply app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java:80]: We don't do it in any of the other functions so I figured it probably didn't matter.

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java (hunk 2 above)
Comment (cdubthecoolcat) [reply app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java]: I had initially tried returning it as a connection but the insertConnection method uses a bunch of strings instead.

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java (hunk 1 above)
Comment (tylerbwong) [reply app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java:80]: I think we should keep it and add it to the methods that don't have it.

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java (hunk 2 above)
Comment (tylerbwong) [reply app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.java]: We should change `insertConnection` to take in a `Connection` then.

File: app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.kt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            contentValues.put(WOL_PORT_COL, connection?.portWol)
+            contentValues.put(DEV_PORT_COL, connection?.portDev)
+            contentValues.put(CITY_COL, connection?.city)
+            contentValues.put(STATE_COL, connection?.state)
+            contentValues.put(COUNTRY_COL, connection?.country)
+            contentValues.put(STATUS_COL, connection?.status)
+            contentValues.put(DATE_COL, connection?.date)
+            database.insertOrThrow(CONNECTIONS_TABLE, null, contentValues)
+            database.close()
+            null
```
Comment (tylerbwong) [inline app/src/main/java/io/awaken/data/database/ConnectionDatabaseHelper.kt]: Can this be removed?

File: app/src/main/java/io/awaken/data/model/Connection.kt [hunk 4]
```diff
@@ -11,17 +11,17 @@ import io.reactivex.Single
...
+        val portWol: String,
         val portDev: String,
         val city: String?,
         val state: String?,
         val country: String?,
         val status: String,
-        val date: String
-) {
+        val date: String,
+        @PrimaryKey @ColumnInfo(name = "id") val id: Int = 0
```
Comment (tylerbwong) [inline app/src/main/java/io/awaken/data/model/Connection.kt]: Why was this moved to the bottom?

File: app/src/main/java/io/awaken/ui/connections/ConnectionViewHolder.kt [hunk 5]
```diff
@@ -30,6 +31,7 @@ class ConnectionViewHolder(
     internal val location = itemView.locationLabel
     internal val date = itemView.awokenDateLabel
     internal val status = itemView.statusMarker
+    private val disposables = CompositeDisposable()
```
Comment (tylerbwong) [inline app/src/main/java/io/awaken/ui/connections/ConnectionViewHolder.kt]: So the reason we use a `CompositeDisposable` is to be able to cancel our long running network requests and prevent leaking resources. For now we can just not use one, but in the future we should move all of the I/O related logic to a `ViewModel`, and call `disposables.clear()` accordingly.

File: app/src/main/java/io/awaken/ui/connections/ConnectionViewHolder.kt [hunk 6]
```diff
@@ -66,19 +68,29 @@ class ConnectionViewHolder(
...
                     refreshListener.invoke()
-                }
-                catch (e: Exception) {
+                } catch (e: Exception) {
                     message = "Failed to delete " + nickname.text.toString()
                     Log.e("failure", e.message)
                 }
 
-                Toast.makeText(itemView.context, message, Toast.LENGTH_LONG).show()
+                Snackbar.make(itemView, message, Snackbar.LENGTH_LONG)
```
Comment (tylerbwong) [inline app/src/main/java/io/awaken/ui/connections/ConnectionViewHolder.kt]: Do we want to show this if the connection wasn't actually deleted? What if `connection` is null?

File: app/src/main/java/io/awaken/data/model/Connection.kt (hunk 4 above)
Comment (cdubthecoolcat) [reply app/src/main/java/io/awaken/data/model/Connection.kt]: So i can use id as an optional argument

Comment (cdubthecoolcat) [reply app/src/main/java/io/awaken/data/model/Connection.kt]: since when initially inserting a connection the database assigns the id

Comment (tylerbwong) [reply app/src/main/java/io/awaken/data/model/Connection.kt]: Oh ok, use `-1` instead.

File: app/src/main/res/xml/network_security_config.xml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <domain-config cleartextTrafficPermitted="true">
+        <domain includeSubdomains="true">api.ipstack.com</domain>
+    </domain-config>
+</network-security-config>
```
Comment (tylerbwong) [inline app/src/main/res/xml/network_security_config.xml]:
```suggestion
</network-security-config>

```

**Review by tylerbwong — APPROVED** [review]
🔥
